pilot/cmd/pilot-agent/main.go

```diff
@@ -335,17 +335,18 @@ func getPilotSAN(domain string, ns string) []string {
 	return pilotSAN
 }
 
-func getDNSDomain(domain string) string {
-	if len(domain) == 0 {
+func getDNSDomain(domain []string) string {
+	out := domain[0]
```

Comment:
why did you change to an array arg if you always take the first element?

Reply:
@ymesika the main reason was to consolidate DNSDomain and the DNSDomains meta as requested in a prior review from @rshriram.  pilot-agent only requires the first entry (what was previously DNSDomain) from the DNSDomains array.

Cheers
-steve

Reply:
@ymesika will take a look at changing getDNSDomain to return an array and make getPilotSan handle the input properly (by returning an array as suggested by @hzxuzhonghu).

Cheers
-steve